src/readii/negative_controls_refactor/regions.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+from scipy.ndimage import binary_dilation
+
+from .abstract_classes import RegionStrategy
+
+
+class FullRegion(RegionStrategy):
+	"""Region strategy to apply control to the entire image."""
+
+	region_name = "full"
+
+	def __call__(self, image_array: np.ndarray, mask_array: np.ndarray) -> np.ndarray:
```

Comment:
This image_array is never used in any of the class call functions. Is it needed?

Reply:
commented below: https://github.com/bhklab/readii/pull/81#issuecomment-2537077528

Reply:
Copying your answer:
> this is a good catch, when I initially defined the base classes I didn't know how the transformations were done and then I didn't know what other regions might be implemented in the future so I left the `image_array` parameter.
> 
> if we aren't looking to implement new regions we can remove the parameter.
> 
> some ideas I had for regions that would use the original `image` would look like :
> 
> ```python
>class ThresholdRegion(RegionStrategy):
> 	"""
> 	Region strategy to include only pixels in the image where intensity exceeds a threshold.
> 
> 	Unlike ROI-based strategies, this strategy does not consider the mask_array.
> 	"""
> 
> 	region_name: str = "threshold"
> 
> 	def __init__(self, threshold: float):
> 		"""
> 		Initialize the region strategy with a given intensity threshold.
> 
> 		Parameters
> 		----------
> 		threshold : float
> 			The intensity threshold for including pixels in the region.
> 		"""
> 		self.threshold = threshold
> 
> 	def __call__(self, image_array: np.ndarray, _: np.ndarray) -> np.ndarray:
> 		"""
> 		Apply the threshold-based region mask to the image.
> 
> 		Parameters
> 		----------
> 		image_array : np.ndarray
> 			The full image array to evaluate pixel intensities.
> 		_ : np.ndarray
> 			Ignored in this implementation.
> 
> 		Returns
> 		-------
> 		np.ndarray
> 			A binary mask including only pixels in the image that exceed the threshold.
> 		"""
> 		# Apply the threshold directly to the image array
> 		region_mask = np.where(image_array > self.threshold, 1, 0)
> 
> 		# Validate that the resulting mask is not empty
> 		if not region_mask.any():
> 			raise ValueError("No pixels in the image exceed the specified intensity threshold.")
> 
> 		return region_mask
> ```


Okay yeah I see how that could be used then. Leave it for now, this might be useful if we start messing around with contrast levels and stuff.